frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/secret.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { SecretData } from '@console/internal/components/configmap-and-secret-data';
+import { k8sGet } from '@console/internal/module/k8s/resource';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { Base64 } from 'js-base64';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { isBound } from '../../utils';
+
+export const GetSecret: React.SFC<GetSecretProps> = ({ obj }) => {
+  const secretData = {
+    BUCKET_NAME: Base64.encode(_.get(obj, 'metadata.name')),
+    ACCESS_KEY: Base64.encode('loading..'),
+    SECRET_KEY: Base64.encode('loading..'),
+    ENDPOINT: Base64.encode('http://loading.com'),
+  };
+  if (isBound(obj)) {
+    const secret = k8sGet(
+      SecretModel,
+      _.get(obj, 'metadata.name'),
+      _.get(obj, 'metadata.namespace'),
+    );
+    const endpoint = k8sGet(
+      ConfigMapModel,
+      _.get(obj, 'metadata.name'),
+      _.get(obj, 'metadata.namespace'),
+    );
+    Promise.all([secret, endpoint])
+      .then((datas) => {
+        secretData.ACCESS_KEY = Base64.encode(_.get(datas[0], 'data.AWS_ACCESS_KEY_ID'));
+        secretData.SECRET_KEY = Base64.encode(_.get(datas[0], 'data.AWS_SECRET_ACCESS_KEY'));
+        secretData.ENDPOINT = `${Base64.encode(_.get(datas[1], 'data.BUCKET_HOST'))}${Base64.encode(
+          `:${_.get(datas[1], 'data.BUCKET_PORT')}`,
+        )}`;
+      })
+      .catch(() => {
+        return null;
```

Comment:
ack

Reply:
I have set it to return `undefined`. The promise is within an if block which checks for the existence of the items that we are querying for and since we have checked that these objects indeed do exist the promise shouldn't fail.